### PR TITLE
feat: angular 15

### PR DESCRIPTION
**Pull request by ronnymikalsen**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalLibraryOfNorway/ngx-mime/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
BREAKING CHANGE: Angular 15

## Does this PR introduce a breaking change?

```
[X] Yes
[ ] No
```

### CODE REVIEW

File: libs/ngx-mime/src/lib/viewer/viewer-footer/canvas-group-navigator/canvas-group-navigator.component.ts [hunk 1]
```diff
@@ -48,7 +48,10 @@ export class CanvasGroupNavigatorComponent implements OnInit, OnDestroy {
       this.iiifManifestService.currentManifest.subscribe(
         (manifest: Manifest | null) => {
           if (manifest) {
-            this.invert = manifest.viewingDirection === ViewingDirection.LTR;
+            this.dir =
+              manifest.viewingDirection === ViewingDirection.LTR
+                ? 'ltr'
+                : 'rtl';
```
Comment (raymondkarstensen) [inline libs/ngx-mime/src/lib/viewer/viewer-footer/canvas-group-navigator/canvas-group-navigator.component.ts:54]: Cannot value from enum of ViewingDirection be used here instead of hardcoding the value in the component?

**Review by raymondkarstensen — CHANGES_REQUESTED** [review]

File: libs/ngx-mime/package.json [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
-    "@angular/core": ">=14.0.0 <15.0.0",
-    "@angular/flex-layout": ">=14.0.0 <15.0.0 || >=14.0.0-beta.41 <15.0.0",
-    "@angular/material": ">=14.0.0 <15.0.0",
-    "d3": ">=5.7.0 <8.0.0",
-    "openseadragon": ">=2.4.0 <4.0.0",
+    "@angular/common": ">=15.0.0 <16.0.0",
+    "@angular/core": ">=15.0.0 <16.0.0",
+    "@angular/flex-layout": ">=15.0.0 <16.0.0 || >=14.0.0-beta.41 <16.0.0",
+    "@angular/material": ">=15.0.0 <16.0.0",
+    "d3": ">=5.7.0 <6.0.0",
```
Comment (jimarthurnilsen) [inline libs/ngx-mime/package.json:9]: Is this peer-dependecy range correct?

File: apps/elements/src/test.ts [hunk 3]
```diff
@@ -1,22 +1,13 @@
...
-  context(
-    path: string,
-    deep?: boolean,
-    filter?: RegExp
-  ): {
-    <T>(id: string): T;
-    keys(): string[];
-  };
-};
+import 'zone.js/testing';
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply apps/elements/src/test.ts]: test.ts is not needed anymore

File: apps/integration-e2e/src/pages/viewer.po.ts [hunk 4]
```diff
@@ -74,13 +74,15 @@ export class ViewerPage {
   private viewMenuCloseButton: Locator;
   private viewMenuDialog: Locator;
   private pageGroup: Locator;
+  private dir: Locator;
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply apps/integration-e2e/src/pages/viewer.po.ts]: I have changed it to viewingDirection

File: libs/ngx-mime/src/lib/content-search-dialog/content-search-dialog-config-strategy.ts [hunk 5]
```diff
@@ -18,6 +17,7 @@ export class MobileContentSearchDialogConfigStrategy
       autoFocus: false,
       width: '100%',
       height: '100%',
+      maxWidth: '100% !important',
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/content-search-dialog/content-search-dialog-config-strategy.ts]: You have right. I have removed it

File: libs/ngx-mime/src/lib/content-search-dialog/content-search-dialog-config-strategy.ts [hunk 6]
```diff
@@ -41,6 +41,7 @@ export class DesktopContentSearchDialogConfigStrategy
       disableClose: false,
       autoFocus: false,
       width: `${DesktopContentSearchDialogConfigStrategy.dialogWidth}px`,
+      maxWidth: '100% !important',
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/content-search-dialog/content-search-dialog-config-strategy.ts]: You have right. I have removed it

File: libs/ngx-mime/src/lib/viewer/viewer-footer/canvas-group-navigator/canvas-group-navigator.component.html [hunk 7]
```diff
@@ -1,16 +1,18 @@
 <mat-toolbar>
   <div fxLayout="row" fxFlex fxLayoutAlign="start center">
-    <div fxFlex>
+    <div fxFlex data-test-id="slider-dir" [dir]="dir">
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/viewer/viewer-footer/canvas-group-navigator/canvas-group-navigator.component.html]: Changed it to 'navigation-slider-container'

File: libs/ngx-mime/src/lib/viewer/viewer-footer/canvas-group-navigator/canvas-group-navigator.component.ts (hunk 1 above)
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/viewer/viewer-footer/canvas-group-navigator/canvas-group-navigator.component.ts]: Yes. I have fixed it

File: libs/ngx-mime/package.json (hunk 2 above)
Comment (ronnymikalsen) [reply libs/ngx-mime/package.json]: I have fixed it

File: libs/ngx-mime/src/lib/view-dialog/view-dialog.component.scss [hunk 8]
```diff
@@ -15,7 +19,11 @@ section {
   margin-left: 16px;
 }
 
-.mat-dialog-content {
+.mat-mdc-dialog-title {
+  padding: 0 2px 16px;
+}
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/view-dialog/view-dialog.component.scss]: I have fixed it

File: libs/ngx-mime/src/lib/viewer/viewer-header/viewer-header.component.ts [hunk 9]
```diff
@@ -71,7 +71,7 @@ export class ViewerHeaderComponent implements OnInit, OnDestroy {
   isContentSearchEnabled = false;
   isFullscreenEnabled = false;
   isInFullscreen = false;
-  fullscreenLabel = this.intl.fullScreenLabel;
+  fullscreenLabel = '';
```
Comment (raymondkarstensen) [inline libs/ngx-mime/src/lib/viewer/viewer-header/viewer-header.component.ts:74]: Why is this removed?

File: libs/ngx-mime/tsconfig.json [hunk 10]
```diff
@@ -10,14 +19,9 @@
...
     }
   ],
-  "compilerOptions": {
-    "forceConsistentCasingInFileNames": true,
-    "strict": true,
-    "noImplicitReturns": true,
-    "noFallthroughCasesInSwitch": true,
-    "target": "es2020"
-  },
+  "extends": "../../tsconfig.base.json",
```
Comment (raymondkarstensen) [inline libs/ngx-mime/tsconfig.json:22]: For readability, isn't it better for this to be in top of file?

File: libs/ngx-mime/tsconfig.lib.json [hunk 11]
```diff
@@ -1,19 +1,17 @@
...
+    "types": []
   },
   "exclude": [
-    "src/test.ts",
-    "**/*.spec.ts",
-    "**/*-test.module.ts",
-    "**/test-*.ts"
+    "src/**/*.spec.ts",
+    "jest.config.ts",
+    "src/**/*.test.ts",
```
Comment (raymondkarstensen) [inline libs/ngx-mime/tsconfig.lib.json:13]: Aren't these test.ts files deleted?

File: libs/ngx-mime/tsconfig.spec.json [hunk 12]
```diff
@@ -1,10 +1,8 @@
...
   "extends": "./tsconfig.json",
   "compilerOptions": {
-    "outDir": "../../dist/out-tsc",
-    "module": "es2020",
+    "outDir": "../../../dist/out-tsc",
     "types": ["jasmine", "node"]
   },
-  "files": ["src/test.ts"],
-  "include": ["**/*.spec.ts", "**/*.d.ts"]
+  "include": ["src/**/*.spec.ts", "src/**/*.test.ts", "src/**/*.d.ts"]
```
Comment (raymondkarstensen) [inline libs/ngx-mime/tsconfig.spec.json:7]: Aren't these test.ts files deleted?

File: libs/ngx-mime/package.json [hunk 13]
```diff
@@ -2,12 +2,12 @@
   "name": "@nationallibraryofnorway/ngx-mime",
   "version": "0.0.1",
   "peerDependencies": {
-    "@angular/common": ">=14.0.0 <15.0.0",
-    "@angular/core": ">=14.0.0 <15.0.0",
-    "@angular/flex-layout": ">=14.0.0 <15.0.0 || >=14.0.0-beta.41 <15.0.0",
-    "@angular/material": ">=14.0.0 <15.0.0",
+    "@angular/common": ">=15.0.0 <16.0.0",
+    "@angular/core": ">=15.0.0 <16.0.0",
+    "@angular/flex-layout": ">=15.0.0 <16.0.0 || >=14.0.0-beta.41 <16.0.0",
```
Comment (raymondkarstensen) [inline libs/ngx-mime/package.json:7]: Should this be bumped to 15.0.0-beta.42?

**Review by raymondkarstensen — CHANGES_REQUESTED** [review]

File: libs/ngx-mime/src/lib/viewer/viewer-header/viewer-header.component.ts (hunk 9 above)
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/viewer/viewer-header/viewer-header.component.ts:74]: ```this.intl``` gives NPE. Initalization moved to ngOnInit

File: apps/demo/src/app/viewer/viewer.component.ts [hunk 14]
```diff
@@ -32,6 +32,8 @@ export class ViewerComponent implements OnDestroy {
     private router: Router,
     private manifestService: ManifestService
   ) {
+    console.log('tester');
+
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply apps/demo/src/app/viewer/viewer.component.ts]: whoopsie ...

**Review by raymondkarstensen — APPROVED** [review]
